pkg/k8s/types.go

```diff
@@ -102,14 +109,20 @@ type Controller interface {
 	// GetEndpoints returns the endpoints for a given service, if found
 	GetEndpoints(service.MeshService) (*corev1.Endpoints, error)
 
-	// UpdateStatus updates the status subresource for the given resource and GroupVersionKind
-	// The object within the 'interface{}' must be a pointer to the underlying resource
-	UpdateStatus(interface{}) (metav1.Object, error)
-
 	// GetPodForProxy returns the pod for the given proxy
 	GetPodForProxy(*envoy.Proxy) (*v1.Pod, error)
 
-	GetTargetPortForServicePort(types.NamespacedName, uint16) (uint16, error)
-
 	ServiceToMeshServices(svc corev1.Service) []service.MeshService
 }
+
+// PassthroughInterface is the interface for methods that are implemented by the k8s.Client, but are not considered
+// k8s "specific", and thus do not need further abstraction, and can be used throughout the code base without fear of
+// coupling to k8s.
+type PassthroughInterface interface {
+	GetMeshConfig() configv1alpha2.MeshConfig
+	GetOSMNamespace() string
+	UpdateIngressBackendStatus(obj *policyv1alpha1.IngressBackend) (*policyv1alpha1.IngressBackend, error)
+	UpdateUpstreamTrafficSettingStatus(obj *policyv1alpha1.UpstreamTrafficSetting) (*policyv1alpha1.UpstreamTrafficSetting, error)
+
+	GetTargetPortForServicePort(types.NamespacedName, uint16) (uint16, error)
```

Comment:
These 3 seem k8s specific to me, in which case they should be a part of the k8s.Controller interface because they are tightly coupled to k8s resources.

What was your reasoning to not add them within the k8s client?

Reply:
The Passthrough interface is part of the k8s Client as well (it's the set of Methods that we allow to "leak" out of k8s)

I'd be happy to abstract them, but not sure the best way to do that at the moment. Open to suggestions though

Reply:
Given these are k8s specific, the comment `but are not considered k8s "specific"` in the description of the PassthroughInterface seems a bit ambiguous. I am trying to understand how do you decide what goes into this interface. A comment describing these criteria in the interface definition would help.

Reply:
added some comments, let me know if that helps